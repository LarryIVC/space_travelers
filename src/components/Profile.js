import { useSelector } from 'react-redux';

const Profile = () => {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div className="my-profile">
      <div>
        <h2>My Missions</h2>
        <ul>
          {joinedMissions.map((mission) => (
            <li key={mission.mission_id}>
              {mission.mission_name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
