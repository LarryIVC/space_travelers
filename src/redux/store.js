import { configureStore } from '@reduxjs/toolkit';
import dragonsReducer from './dragons/dragonsSlice';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    dragons: dragonsReducer,
    missions: missionsReducer,
  },
});

export default store;
