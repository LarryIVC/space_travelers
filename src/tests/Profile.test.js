import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Profile from "../components/Profile";
import "@testing-library/jest-dom/extend-expect";
import store from "../redux/store";
import { Provider } from "react-redux";

describe("Test suite for the Profile component", () => {
  let profile;
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Profile />
        </Provider>
      </BrowserRouter>
    );
    profile = screen.getByText(/My Missions/i);
  });

  it("Should render a snapshot of the Profile component", () => {
    expect(profile).toMatchSnapshot();
  });
});
