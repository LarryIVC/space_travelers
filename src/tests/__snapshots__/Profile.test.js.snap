// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Profile component Should render a snapshot of the Profile component 1`] = `
<h2>
  My Missions
</h2>
`;
